src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -838,28 +840,35 @@ private List<GrantedAuthority> determineAuthorities(IdToken idToken, GenericJson
         grantedAuthorities.add(SecurityRealm.AUTHENTICATED_AUTHORITY2);
 
         if (isNotBlank(groupsFieldName)) {
-            if (!Strings.isNullOrEmpty(userInfoServerUrl) && containsField(userInfo, groupsFieldName)) {
-                LOGGER.fine("UserInfo contains group field name: " + groupsFieldName + " with value class:" + getField(userInfo, groupsFieldName).getClass());
-                List<String> groupNames = ensureString(getField(userInfo, groupsFieldName));
+            // if groupsFieldName contains []., then groupsFieldName
+            // is first portion, and nestedGroupFieldName is
+            // second portion
+            String[] parts = groupsFieldName.split("\\[\\]\\.");
```

Comment:
use limit to limit splitting to 2 `String[] split(String regex, int limit)` - granted there should be only one and warning should be generated if format is not correct.

Note: a method `doCheckGroupsFieldName` should be added to check field format for GUI users.
If you don't do it, I'll make it is another PR.

Reply:
updated in latest changes

Reply:
added doCheckGroupsFieldName() to check format